File: compile.clj
#!/usr/bin/env lein-exec

(def PRO "cgraph")
(def DIR ".")
(def INC (str DIR "/" "include"))
(def SRC (str DIR "/" "src"))
(def TST (str DIR "/" "test"))
(def LIB (str DIR "/" "lib"))

(def CC "cc")
(def CFLAGS "-pedantic -Wall -fpic -std=c89")
(def CSFLAGS "-shared")

(def MODE "debug")
(def CFLAGS (if (compare MODE "debug") (str CFLAGS " -g -DDEBUG") (if (compare MODE "release") (str CFLAGS " -static -O2"))))

(def AR "ar")
(def ARFLAGS "-rcs")

File: compile.scala
#!/bin/sh
exec scala "$0" "$@"
!#

import scala.reflect.io.Directory

object Compile {
  def main(args: Array[String]): Unit = {
    // project
    val PRO : String = "cgraph"

    // directory tree
    val DIR : String = "."
    val INC : String = "$DIR/include"
    val SRC : String = "$DIR/src"
    val TST : String = "$DIR/test"
    val LIB : String = "$DIR/lib"

    val CC : String = "cc"
    var CFLAGS : String = "-pedantic -Wall -fpic -std=c89"
    val CSFLAGS : String = "-shared"

    val MODE : String = "debug"
    if (MODE.equals("debug"))
    {
      CFLAGS += " -g -DDEBUG"
    }
    else if (MODE.equals("release"))
    {
      CFLAGS += " -static -O2"
    }

    val AR : String = "ar"
    val ARFLAGS : String = "-rcs"
  }
}